internal/admin/snwatcher/snwatcher.go

```diff
@@ -121,60 +118,69 @@ func (snw *StorageNodeWatcher) checkHeartbeat(ctx context.Context, now time.Time
 		go func(idx int) {
 			defer wg.Done()
 			snid := md.StorageNodes[idx].StorageNodeID
-			snmd, err := snw.snmgr.GetMetadata(ctx, snid)
+
+			mctx, cancel := context.WithTimeout(ctx, snw.heartbeatCheckDeadline)
+			defer cancel()
```

Comment:
mctx can be shared by all Go routines. It is safe as well as reduces heap allocations; RPC failure of a goroutine can't affect other goroutines' mctx use.

Reply:
You're right. Fixed.